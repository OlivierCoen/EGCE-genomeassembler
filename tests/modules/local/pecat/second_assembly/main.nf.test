nextflow_process {

    name "Test Process PECAT_SECOND_ASSEMBLY"
    script "modules/local/pecat/second_assembly/main.nf"
    process "PECAT_SECOND_ASSEMBLY"
    tag "pecat_second_assembly"

    test("Normal fasta") {

        tag "pecat_second_assembly_normal"

        when {

            process {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true),
                    file( '$projectDir/tests/test_data/pecat/output/normal/phase/results.tar.gz', checkIfExists: true)
                ]
                input[1] = file( '$projectDir/tests/test_data/pecat/input/pecat_optional_config.txt', checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Small fasta") {

        tag "pecat_second_assembly_small"

        when {

            process {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true),
                    file( '$projectDir/tests/test_data/pecat/output/small/phase/results.tar.gz', checkIfExists: true)
                ]
                input[1] = file( '$projectDir/tests/test_data/pecat/input/pecat_optional_config.txt', checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
