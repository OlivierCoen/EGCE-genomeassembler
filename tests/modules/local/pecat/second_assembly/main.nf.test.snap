{
    "Small fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,7d9f7677e55ba76733ac82231a6a0fc5"
                    ]
                ],
                "1": [
                    "versions.yml:md5,988e86b1280bf7e6e61097e8277e9340"
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,7d9f7677e55ba76733ac82231a6a0fc5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,988e86b1280bf7e6e61097e8277e9340"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-24T20:37:57.604054067"
    },
    "Normal fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,19d3ec91b6256a76bb639a64bf8a5dfa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,988e86b1280bf7e6e61097e8277e9340"
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "results.tar.gz:md5,19d3ec91b6256a76bb639a64bf8a5dfa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,988e86b1280bf7e6e61097e8277e9340"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-24T20:41:00.037130986"
    }
}